db/migrate/20190802133303_create_triggers_backend_apis.rb

```diff
@@ -0,0 +1,204 @@
+# frozen_string_literal: true
+
+class CreateTriggersBackendApis < ActiveRecord::Migration
+  # At this point it does not matter if it blocks these tables or not because they are not being used yet anyway
+
+  def up
+    BackendApi.transaction do
+      my_class = self.class
+
+      my_class.drop_triggers
+      my_class.create_triggers
+    end
+  end
+
+  def down
+    BackendApi.transaction do
+      self.class.drop_triggers
+    end
+  end
+
+  def self.drop_triggers
+    %w[backend_apis backend_api_configs].each do |table_name|
+      public_send("drop_trigger_#{System::Database.adapter}", table_name)
+    end
+  end
+
+  def self.create_triggers
+    %w[backend_apis backend_api_configs].each do |table_name|
+      public_send("create_trigger_#{table_name}_#{System::Database.adapter}")
+    end
+  end
+
+  def self.drop_trigger_mysql(table_name)
+    safety_assured do
+      execute <<~SQL.strip
+        DROP TRIGGER IF EXISTS #{table_name}_tenant_id;
+      SQL
+    end
+  end
+
+  def self.drop_trigger_postgres(table_name)
+    safety_assured do
+      execute <<~SQL.strip
+        DROP TRIGGER IF EXISTS #{table_name}_tenant_id ON #{table_name};
+      SQL
+      execute <<~SQL.strip
+        DROP FUNCTION IF EXISTS tp_#{table_name}_tenant_id;
+      SQL
+    end
+  end
+
+  def self.drop_trigger_oracle(table_name)
+    safety_assured do
+      execute <<~SQL.strip
+        BEGIN
+        EXECUTE IMMEDIATE 'DROP TRIGGER #{table_name}_tenant_id';
+        EXCEPTION
+        WHEN OTHERS THEN
+         IF SQLCODE != -4080 THEN
+           RAISE;
+         END IF;
+        END;
+      SQL
+    end
+  end
+
+  def self.create_trigger_backend_apis_mysql
+    safety_assured do
+      execute <<~SQL.strip
+        CREATE TRIGGER backend_apis_tenant_id BEFORE INSERT
```

Comment:
We should be using https://github.com/3scale/porta/blob/master/lib/system/database/trigger.rb#L5 and https://github.com/3scale/porta/blob/master/lib/system/database/mysql/trigger.rb#L6 for this.